src/hooks/useSeen.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useRef } from 'react';
+
+import { selectRawIndexerHeightData } from '@/abacus-ts/selectors/base';
+import { shallowEqual } from 'react-redux';
+
+import { getUserWalletAddress } from '@/state/accountSelectors';
+import { setSeenFills, setSeenOpenOrders, setSeenOrderHistory } from '@/state/accountUiMemory';
+import { getSelectedNetwork } from '@/state/appSelectors';
+import { useAppDispatch, useAppSelector } from '@/state/appTypes';
+
+import { assertNever } from '@/lib/assertNever';
+
+export function useViewPanel(
+  market: string | undefined,
+  kind: 'fills' | 'openOrders' | 'orderHistory'
+) {
+  const networkId = useAppSelector(getSelectedNetwork);
+  const walletId = useAppSelector(getUserWalletAddress);
+  const height = useAppSelector(selectRawIndexerHeightData);
+  const lastSetCore = useRef<any[]>([]);
+
+  const dispatch = useAppDispatch();
+  const actionCreator =
+    kind === 'fills'
+      ? setSeenFills
+      : kind === 'openOrders'
+        ? setSeenOpenOrders
+        : // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          kind === 'orderHistory'
+          ? setSeenOrderHistory
+          : assertNever(kind);
+
+  const componentWillUnmount = useComponentWillUnmount();
+
+  useEffect(() => {
+    if (height != null && walletId != null) {
+      // only set once for a given set of configurations
+      // effectively, view on mount as soon as we load height
+      const thisCore = [market, actionCreator, networkId, walletId];
+      if (!shallowEqual(lastSetCore.current, thisCore)) {
+        lastSetCore.current = thisCore;
+        dispatch(actionCreator({ scope: { networkId, walletId }, market, height }));
```

Comment:
height has a refreshTime of ~ 10sec (~10 blocks)? Would this affect how we do comparisons at all? Especially if we are 9 blocks past the latest fetched blockHeight and that latest fetch is what we are saving.

Reply:
Yeah I thought about this and I don't really have an answer for it other than hoping it doesn't happen much. Basically if you get an update from the websocket and then flip over to view it within ~5 ish seconds then marking it viewed won't actually mark it viewed.